codeflash/discovery/discover_unit_tests.py

```diff
@@ -37,13 +39,100 @@ class TestFunction:
 FUNCTION_NAME_REGEX = re.compile(r"([^.]+)\.([a-zA-Z0-9_]+)$")
 
 
+class TestsCache:
+    def __init__(self) -> None:
+        self.connection = sqlite3.connect(codeflash_cache_dir / "tests_cache.db")
+        self.cur = self.connection.cursor()
+
+        self.cur.execute(
+            """
+            CREATE TABLE IF NOT EXISTS discovered_tests(
+                file_path TEXT,
+                file_hash TEXT,
+                qualified_name_with_modules_from_root TEXT,
+                function_name TEXT,
+                test_class TEXT,
+                test_function TEXT,
+                test_type TEXT,
+                line_number INTEGER,
+                col_number INTEGER
+            )
+            """
+        )
+        self.cur.execute(
+            """
+            CREATE INDEX IF NOT EXISTS idx_discovered_tests_file_path_hash
+            ON discovered_tests (file_path, file_hash)
+            """
+        )
+        self._memory_cache = {}
+
+    def insert_test(
+        self,
+        file_path: str,
+        file_hash: str,
+        qualified_name_with_modules_from_root: str,
+        function_name: str,
+        test_class: str,
+        test_function: str,
+        test_type: TestType,
+        line_number: int,
+        col_number: int,
+    ) -> None:
+        test_type_value = test_type.value if hasattr(test_type, "value") else test_type
+        self.cur.execute(
+            "INSERT INTO discovered_tests VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)",
+            (
+                file_path,
+                file_hash,
+                qualified_name_with_modules_from_root,
+                function_name,
+                test_class,
+                test_function,
+                test_type_value,
+                line_number,
+                col_number,
+            ),
+        )
+        self.connection.commit()
+
+    def get_tests_for_file(self, file_path: str, file_hash: str) -> list[FunctionCalledInTest]:
+        cache_key = (file_path, file_hash)
+        if cache_key in self._memory_cache:
+            return self._memory_cache[cache_key]
+        self.cur.execute("SELECT * FROM discovered_tests WHERE file_path = ? AND file_hash = ?", (file_path, file_hash))
+        result = [
+            FunctionCalledInTest(
+                tests_in_file=TestsInFile(
+                    test_file=Path(row[0]), test_class=row[4], test_function=row[5], test_type=TestType(int(row[6]))
+                ),
+                position=CodePosition(line_no=row[7], col_no=row[8]),
+            )
+            for row in self.cur.fetchall()
+        ]
+        self._memory_cache[cache_key] = result
+        return result
+
+    @staticmethod
+    def compute_file_hash(path: str) -> str:
+        h = hashlib.md5(usedforsecurity=False)
```

Comment:
I'm fine with this but I've seen some orgs prefer sha256 as its data integrity hash. its mostly for security reasons though, although 256 bits almost eliminates hash collisions over 128 bit hash.
https://help.ubuntu.com/community/HowToSHA256SUM
https://security.stackexchange.com/questions/203344/how-likely-is-a-collision-using-md5-compared-to-sha256-for-checking-file-integr

Reply:
i think its a good idea to standardize on sha256 for all data integrity hashing with codeflash